items.go

```diff
@@ -127,14 +127,47 @@ func (p *Pokemon) UseMedicine(i *Item) (t []Transaction) {
 func (p *Pokemon) UseBerryInAPinch(i *Item) (t []Transaction) {
 	switch i.ID {
 	case ItemApicotBerry:
+		t = append(t, ModifyStatTransaction{
+			Target: p,
+			Stat:   StatSpDef,
+			Stages: 1,
+		})
 	case ItemCustapBerry:
+		// TODO: Force pokemon to go first
 	case ItemGanlonBerry:
+		t = append(t, ModifyStatTransaction{
+			Target: p,
+			Stat:   StatDef,
+			Stages: 1,
+		})
 	case ItemLansatBerry:
+		t = append(t, ModifyStatTransaction{
+			Target: p,
+			Stat:   StatCritChance,
+			Stages: 2,
+		})
 	case ItemLiechiBerry:
+		t = append(t, ModifyStatTransaction{
+			Target: p,
+			Stat:   StatAtk,
+			Stages: 1,
+		})
 	case ItemMicleBerry:
+		// TODO: Perfect accuracy for next move
 	case ItemPetayaBerry:
+		t = append(t, ModifyStatTransaction{
+			Target: p,
+			Stat:   StatSpAtk,
+			Stages: 1,
+		})
 	case ItemSalacBerry:
+		t = append(t, ModifyStatTransaction{
+			Target: p,
+			Stat:   StatSpeed,
+			Stages: 1,
+		})
 	case ItemStarfBerry:
+		// TODO: boost random stat, requires battle RNG to be available.
```

Comment:
Do you have ideas on whether we should pass in the battle object, BattleContext, or something else?

Reply:
Honestly, I'd prefer to move it all back into the transaction's `Mutate()`

Reply:
I agree, I had implemented that system for items before the transaction system was well defined.